src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -5671,16 +5677,11 @@ internal Ast CommandRule(bool forDynamicKeyword)
                         case TokenKind.Semi:
                         case TokenKind.AndAnd:
                         case TokenKind.OrOr:
+                        case TokenKind.Ampersand:
```

Comment:
>TokenKind [](start = 29, length = 9)

what do the following emit?
`$A = "bar" & "foo"`
or
`$A = $("bar" & echo "foo")`
and we should probably have tests for them